sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -257,11 +256,24 @@ class Analyzer(catalog: Catalog,
 
       case q: LogicalPlan =>
         logTrace(s"Attempting to resolve ${q.simpleString}")
-        q transformExpressionsUp  {
+        q transformExpressionsUp {
           case u @ UnresolvedAttribute(name) if resolver(name, VirtualColumn.groupingIdName) &&
             q.isInstanceOf[GroupingAnalytics] =>
             // Resolve the virtual column GROUPING__ID for the operator GroupingAnalytics
             q.asInstanceOf[GroupingAnalytics].gid
+          case u @ UnresolvedAttribute(name) if q.isInstanceOf[Sort] =>
+            val s = q.asInstanceOf[Sort]
+            val newChild = s.child match {
+              case Project(list, c) =>
+                val newList = list.filter {
+                  case Alias(g: GetField, _) => false
+                  case Alias(g: GetItem, _) => false
+                  case _ => true
+                }
+                Project(newList, c)
+              case other => other
+            }
+            Sort(s.order, s.global, newChild).resolveChildren(name, resolver).getOrElse(u)
```

Comment:
I think it will not work when we have a order by in a subquery and the outer query block try to access the those filtered out fields.